plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/OpenShiftServerAdapterFactory.java

```diff
@@ -111,5 +116,44 @@ public Object loadAdapter(Class adapter, IProgressMonitor monitor) {
 		}
 
 	}
+	
```

Comment:
I'm really confused by this. What's the goal here? Why are we adapting to stubs?  If a wst.web project already exists, wouldn't it be better to get it's ACTUAL IWebModule from org.eclipse.jst.server.core  ??